app/src/main/java/com/humberto/tasky/agenda/presentation/agenda_details/AgendaDetailsScreen.kt

```diff
@@ -0,0 +1,148 @@
+package com.humberto.tasky.agenda.presentation.agenda_details
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.humberto.tasky.R
+import com.humberto.tasky.agenda.presentation.AgendaItemType
+import com.humberto.tasky.core.presentation.designsystem.TaskyTheme
+import com.humberto.tasky.core.presentation.designsystem.components.TaskyScaffold
+import com.humberto.tasky.core.presentation.designsystem.components.TaskyToolbar
+import com.humberto.tasky.core.presentation.ui.toFormattedUppercaseDateTime
+import com.humberto.tasky.main.navigation.AgendaDetails
+import java.time.ZonedDateTime
+
+@Composable
+fun AgendaDetailsScreenRoot(
+    agendaDetails: AgendaDetails,
+    onBackClick: () -> Unit,
+    viewModel: AgendaDetailsViewModel = hiltViewModel()
+) {
+    LaunchedEffect(agendaDetails) {
+        viewModel.setStateFromNav(agendaDetails)
+    }
+    val state by viewModel.agendaDetailsState.collectAsState()
+    state?.let {
```

Comment:
Hey guys, what do you think about having this nullable state? Seems weird but it also didn't seem right to have a default agendaItemType in the state.

Reply:
At the moment of navigation, you know which item type you're dealing with since you passed all that as static navigation arguments. The issue in your code is just that you try to pass the item type from the UI to the ViewModel, so there is a delay. With `SavedStateHandle` the ViewModel can instantly get the agenda item type once it's created:
```
class AgendaDetailsViewModel @Inject constructor(
    savedStateHandle: SavedStateHandle,
    ...
): ViewModel() {

    private val agendaItemType: AgendaItemType = savedStateHandle.get<AgendaItemType>("agendaItemType")!! // save to have this assertion here, unless you intentionally pass invalid data
    
    private val _state = MutableStateFlow(
        AgendaDetailState(/* Construct default state with agendaItemType property */)
    )
}
```